frost-core/src/traits.rs

```diff
@@ -258,94 +263,115 @@ pub trait Ciphersuite: Copy + Clone + PartialEq + Debug {
+    /// to ensure the signature is valid under BIP340; for all other ciphersuites
+    /// this simply returns a [`Signature`] wrapping `R` and `z`.
+    fn aggregate_sig_finalize(
         z: <<Self::Group as Group>::Field as Field>::Scalar,
-        challenge: &Challenge<Self>,
-        verifying_key: &Element<Self>,
-    ) -> <<Self::Group as Group>::Field as Field>::Scalar {
-        panic!("Not implemented");
+        R: Element<Self>,
+        _verifying_key: &VerifyingKey<Self>,
+        _msg: &[u8],
+    ) -> Signature<Self> {
+        Signature { R, z }
     }
 
-    /// tweaked z for SigningKey sign (used in frost-sepc256k1-tr)
-    #[allow(unused)]
-    fn tweaked_z(
+    /// Finalize and output a single-signer Schnorr signature.
+    fn single_sig_finalize(
         k: <<Self::Group as Group>::Field as Field>::Scalar,
+        R: Element<Self>,
         secret: <<Self::Group as Group>::Field as Field>::Scalar,
-        challenge: <<Self::Group as Group>::Field as Field>::Scalar,
-        verifying_key: &Element<Self>,
-    ) -> <<Self::Group as Group>::Field as Field>::Scalar {
-        panic!("Not implemented");
+        challenge: &Challenge<Self>,
+        _verifying_key: &VerifyingKey<Self>,
+    ) -> Signature<Self> {
+        let z = k + (challenge.0 * secret);
+        Signature { R, z }
     }
 
-    /// signature_share compatible with taproot (used in frost-sepc256k1-tr)
-    #[allow(unused)]
-    fn compute_taproot_compat_signature_share(
-        signer_nonces: &crate::round1::SigningNonces<Self>,
-        binding_factor: crate::BindingFactor<Self>,
-        group_commitment: crate::GroupCommitment<Self>,
+    /// Compute the signature share for a particular signer on a given challenge.
+    fn compute_signature_share(
+        signer_nonces: &round1::SigningNonces<Self>,
+        binding_factor: BindingFactor<Self>,
+        _group_commitment: GroupCommitment<Self>,
         lambda_i: <<Self::Group as Group>::Field as Field>::Scalar,
-        key_package: &crate::keys::KeyPackage<Self>,
+        key_package: &KeyPackage<Self>,
         challenge: Challenge<Self>,
-    ) -> crate::round2::SignatureShare<Self> {
-        panic!("Not implemented");
+    ) -> round2::SignatureShare<Self> {
+        round2::compute_signature_share(
+            signer_nonces,
+            binding_factor,
+            lambda_i,
+            key_package,
+            challenge,
+        )
     }
 
-    /// calculate tweaked public key (used in frost-sepc256k1-tr)
-    #[allow(unused)]
-    fn tweaked_public_key(
-        public_key: &<Self::Group as Group>::Element,
+    /// Compute the effective group element which should be used for signature operations
+    /// for the given verifying key.
+    ///
+    /// In frost-sepc256k1-tr, this is used to commit the key to taptree merkle root hashes.
+    /// For all other ciphersuites, this simply returns `verifying_key.to_element()`
```

Comment:
You might not want to leave this message here, as this may not be true for all future ciphersuites other than tr

Reply:
Good point. CC @zebra-lucky. You may want to consider removing the 'For all other ciphersuites' comments i added on the default trait methods here.

Reply:
Ok, I'll delete this in separate commit to the upstream branch PR

Reply:
Fixed in:
https://github.com/ZcashFoundation/frost/pull/584/commits/4c849aa2973efe8cd12cd5b629bd48266baf57d5